libs/@blockprotocol/graph/src/codegen/parameters.ts

```diff
@@ -0,0 +1,241 @@
+import {
+  validateVersionedUrl,
+  VersionedUrl,
+} from "@blockprotocol/type-system/slim";
+
+import { typedEntries, typedKeys } from "../shared/util/typed-object-iter.js";
+
+/**
+ * The input parameters of Codegen, prior to their validation.
+ */
+export type CodegenParameters = {
+  /** Files, and the types to generate within them */
+  targetRoot: string;
+  targets: {
+    [fileName: string]: [
+      {
+        versionedUrl: VersionedUrl;
+        /** Whether to generate helper types which mark this Entity Type as the Block Entity */
+        blockEntity?: boolean;
+      },
+    ];
+  };
+  typeNameOverrides?: {
+    [versionedUrl: string]: string;
+  };
+  /** Generate look-up maps with aliases for all type URLs */
+  versionedUrlAliases?:
+    | { enabled: false }
+    | {
+        enabled: true;
+        /** Override the generated name for certain aliases */
+        overrides?: {
+          [versionedUrl: string]: string;
+        };
+      };
+  /* @todo - Add support for generating Base URL aliases? */
+  /* @todo - Make the resolve depth configurable */
+  temporal?: boolean;
+};
+
+export const validateCodegenParameters = (
```

Comment:
This is pretty horrible but the error reports are nice.. perhaps we can just leverage some codegen utility to generate the shape and appropriate type-guard